interview-questions.md

```diff
@@ -0,0 +1,58 @@
+
+First, without external resources. Challenge yourself to answer from memory as if you were in a job interview.
+
+Then, research the question to expand on your answer. Even if you feel you have answered the question completely, there is always something more to learn. Write your researched answer in your OWN WORDS.
+
+1. What is JSX? What is one syntax difference between HTML and JSX?
+
+Your answer: JSX is a way to write HTML-like code in JavaScript and is commonly used in React. One syntax difference between HTML and JSX is the casing. onclick is HTML and onClick is JSX
+
+Researched answer: JSX is a syntax extension for JavaScript that allows you to write HTML-like code within your JavaScript code, primarily used in React applications to describe the structure and content of components. It lets you combine HTML and JavaScript into a single file, making your code more modular and easier to manage. The main syntax difference between HTML and JSX is the way attributes are defined. In HTML, attributes are defined using the regular HTML syntax, where you have the attribute name followed by an equals sign, then the value in quotes.
+
+1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
+
+Your answer: Yarn is a package manager for JavaScript that makes it easy to manage dependencies in your projects. When you run the yarn command in your terminal, the yarn.lock file and the node_modules directory may be modified or created.
+
+Researched answer: Yarn is a package manager for JavaScript that was created by Facebook to address the limitations with the existing npm package manager. Yarn provides faster and more reliable package installation, better support for versioning and caching, and better security. 
+
+3. What is a React component? 
+
+Your answer: A React component is a self encapsulated function that returns JSX. A React component can utilize different hooks from React to add functionality to the component. The useState hook is used to store information. When that information is changed, the component is rerendered to display those changes. 
+
+Researched answer:  React components are independent and reusable bits of code. They serve the same purpose as JavaScript functions, but work in isolation and return HTML. Components come in two types, Class components and Function components.
+
+4. What is the difference between state values and props in React?
+
+Your answer: the difference between state values and props in React are that state values stores the data that determines how a component is displayed to the user. Props is short for properties in React and is just an object that has no value
+
+Researched answer: The difference between state values and props in React is that state values are special variables that store data and props are special React objects that pass information from one component to another. State values define how a component is displayed and behaves.
```

Comment:
Exactly, and props can pass data as well as functions from a parent to a child component. 👏🏽